### PR TITLE
Add event for conditioned transition

### PR DESCRIPTION
**Pull request by danloa**
Resolves #180 

Every time that a conditioned transition is evaluated to true, an event informing the flow, transition and instance is sent to the bus.

### CODE REVIEW

File: src/ProcessMaker/Nayra/Bpmn/TransitionTrait.php [hunk 1]
```diff
@@ -97,6 +98,13 @@ protected function conditionIsFalse()
      */
     protected function doTransit(CollectionInterface $consumeTokens, ExecutionInstanceInterface $executionInstance)
     {
+        if (get_class($this) === ConditionedExclusiveTransition::class && !empty($this->getOwner()->getOwnerDocument())) {
+            $source = $this->outgoing()->item(0)->origin()->getOwner();
+            $target = $this->outgoing->item(0)->target()->getOwner();
+            $flow = $this->findFlow($source, $target);
```
Comment (caleeli) [inline src/ProcessMaker/Nayra/Bpmn/TransitionTrait.php:104]: Could replace the findFlow method by:
```
            $flow = $source->getOutgoingFlows()->findFirst(function (FlowInterface $flow) use ($target) {
                return $flow->getTarget() === $target;
            });
```

File: src/ProcessMaker/Nayra/Bpmn/TransitionTrait.php [hunk 2]
```diff
@@ -120,6 +128,39 @@ protected function doTransit(CollectionInterface $consumeTokens, ExecutionInstan
...
     }
 
+    /**
+     * Find the Node's flow that is traversed
+     * @param $origin
+     * @param $target
+     * @return mixed
+     * @throws \ErrorException
+     */
+    private function findFlow($origin, $target)
```
Comment (caleeli) [inline src/ProcessMaker/Nayra/Bpmn/TransitionTrait.php:138]: This method could be removed

File: src/ProcessMaker/Nayra/Storage/BpmnDocument.php [hunk 3]
```diff
@@ -1120,4 +1120,11 @@ public function getValidationErrors()
     {
         return $this->validationErrors;
     }
+
```
Comment (caleeli) [inline src/ProcessMaker/Nayra/Storage/BpmnDocument.php:1123]: This method could be removed

File: src/ProcessMaker/Nayra/Bpmn/TransitionTrait.php [hunk 4]
```diff
@@ -97,6 +98,13 @@ protected function conditionIsFalse()
      */
     protected function doTransit(CollectionInterface $consumeTokens, ExecutionInstanceInterface $executionInstance)
     {
+        if (get_class($this) === ConditionedExclusiveTransition::class && !empty($this->getOwner()->getOwnerDocument())) {
```
[the remark this answers is not in the document]
Comment (danloa) [reply src/ProcessMaker/Nayra/Bpmn/TransitionTrait.php:101]: done

File: src/ProcessMaker/Nayra/Bpmn/TransitionTrait.php (hunk 2 above)
Comment (danloa) [reply src/ProcessMaker/Nayra/Bpmn/TransitionTrait.php:138]: done

File: src/ProcessMaker/Nayra/Storage/BpmnDocument.php (hunk 3 above)
Comment (danloa) [reply src/ProcessMaker/Nayra/Storage/BpmnDocument.php]: done

File: src/ProcessMaker/Nayra/Bpmn/TransitionTrait.php [hunk 5]
```diff
@@ -97,6 +98,19 @@ protected function conditionIsFalse()
      */
     protected function doTransit(CollectionInterface $consumeTokens, ExecutionInstanceInterface $executionInstance)
     {
+        if ($this instanceof ConditionedExclusiveTransition
+            && !empty($this->getOwner()->getOwnerDocument())) {
```
Comment (caleeli) [inline src/ProcessMaker/Nayra/Bpmn/TransitionTrait.php:102]: I think this condition is not required anymore:

`!empty($this->getOwner()->getOwnerDocument())`

Comment (danloa) [reply src/ProcessMaker/Nayra/Bpmn/TransitionTrait.php]: Yes, the condition it isn't necessary now. It was removed

**Pull request merged**